tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -170,6 +177,125 @@ func (rb *IptablesRuleBuilder) buildRules(rules []*Rule) [][]string {
 	return output
 }
 
+func reverseRules(rules []*Rule) []*Rule {
```

Comment:
Can we add comments? not great to have esoteric parsing logic without explaining what it does

Reply:
:+1: I also asked for more explicit/behaviorally-descriptive unit tests around these mutative funcs, it should be easy to fence these very tightly as there are no OS deps.

Reply:
I added comments to the reverse and check logic, same for the higher-level functions verifying the state of iptables.
Fine-grained tests for reverse and check were also added in iptables_builder_test.go